airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -682,6 +682,7 @@ async def _load_config(self):
 
         if self.config_dict:
             self.log.debug(LOADING_KUBE_CONFIG_FILE_RESOURCE.format("config dictionary"))
+            self._is_in_cluster = False
```

Comment:
Change looks good, but can you please add a unit test, either a new one or in the existing one?